hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestAbfsInputStreamReadFooter.java

```diff
@@ -167,24 +214,55 @@ public void testSeekToEndAndReadWithConfFalse() throws Exception {
 
   private void testSeekAndReadWithConf(boolean optimizeFooterRead,
       SeekTo seekTo) throws Exception {
+    int fileIdx = 0;
+    List<Future> futureList = new ArrayList<>();
+    for (int j = 0; j <= 4; j++) {
+      final int fileSize = (int) Math.pow(2, j) * SIZE_256_KB;
+      final int fileId = fileIdx++;
+      futureList.add(executorService.submit(() -> {
+        try {
+          try (AzureBlobFileSystem spiedFs = createSpiedFs(
+              getRawConfiguration())) {
+            String fileName = methodName.getMethodName() + fileId;
+            byte[] fileContent = getRandomBytesArray(fileSize);
+            Path testFilePath = createFileWithContent(spiedFs, fileName,
+                fileContent);
+            for (int i = 0; i <= 4; i++) {
```

Comment:
I was wondering that since we are now changing the read buffer size as well.
Do we really need file size and read buffer size to have 5 different values...

Objective here is to test all possible scenarios of relative values of three sizes (File size, Read buffer size, Footer read buffer size)
So, permutations of three values of three sizes can be used to achieve that. 
All the three sizes can range in set [256KB, 512KB, 1MB] and we should be good to go.
Earlier file sizes were made to go till 4MB because read buffer size was kept at default 4MB.
This will reduce the test runtime as well.

Reply:
ReadBufferSize has default equal to 4 MB. Will keep this in range. Though have removed 2 MB from readBufferSize and fileSize. Following are the ranges:

fileSize: 256 KB, 512 KB, 1 MB, 4 MB
readBufferSize: 256 KB, 512 KB, 1 MB, 4 MB
footerReadBufferSize: 256 KB, 512 KB, 1 MB,